File: frontend/src/api/documents.ts
import apiClient from './client'

export interface DocumentItem {
  id: string
  title: string
  filename: string
  file_type: string
  file_size_bytes: number
  status: string
  authors: string | null
  journal: string | null
  doi: string | null
  abstract: string | null
  publication_date: string | null
  chunk_count: number
  knowledge_base_id: string
  created_at: string
  error_message: string | null
}

export interface DocumentListResponse {
  items: DocumentItem[]
  total: number
  page: number
  page_size: number
}

export interface DocumentContentResponse {
  id: string
  title: string
  authors: string | null
  abstract: string | null
  content: string
  status: string
  publication_date: string | null
}

export const documentsApi = {
  upload: (file: File, knowledgeBaseId: string, title?: string) => {
    const formData = new FormData()
    formData.append('file', file)
    formData.append('knowledge_base_id', knowledgeBaseId)
    if (title) formData.append('title', title)
    return apiClient.post<DocumentItem>('/documents/upload', formData, {
      headers: { 'Content-Type': 'multipart/form-data' },
    })
  },

  list: (params?: { knowledge_base_id?: string; status?: string; page?: number; page_size?: number }) =>
    apiClient.get<DocumentListResponse>('/documents', { params }),

  get: (id: string) => apiClient.get<DocumentItem>(`/documents/${id}`),

  getContent: (id: string) => apiClient.get<DocumentContentResponse>(`/documents/${id}/content`),

  reindex: (id: string) => apiClient.post(`/documents/${id}/reindex`),

  reindexBatch: (params?: { knowledge_base_id?: string; limit?: number }) =>
    apiClient.post('/documents/reindex', null, { params }),

  delete: (id: string) => apiClient.delete(`/documents/${id}`),
}

File: backend/meks/api/schemas/documents.py
from datetime import date, datetime
from uuid import UUID

from pydantic import BaseModel, field_serializer


class DocumentResponse(BaseModel):
    id: UUID
    title: str
    filename: str
    file_type: str
    file_size_bytes: int
    status: str
    authors: str | None
    journal: str | None
    doi: str | None
    publication_date: date | None
    abstract: str | None
    keywords: str | None
    chunk_count: int
    knowledge_base_id: UUID
    uploaded_by: UUID
    created_at: datetime
    error_message: str | None = None
    analysis_status: str | None = None
    analysis_risk_score: int | None = None
    risk_level: str | None = None

    model_config = {"from_attributes": True}

    @field_serializer("id", "knowledge_base_id", "uploaded_by")
    def serialize_uuid(self, v: UUID) -> str:
        return str(v)

    @field_serializer("created_at")
    def serialize_dt(self, v: datetime) -> str:
        return v.isoformat()

    @field_serializer("publication_date")
    def serialize_date(self, v: date | None) -> str | None:
        return v.isoformat() if v else None


class DocumentListResponse(BaseModel):
    items: list[DocumentResponse]
    total: int
    page: int
    page_size: int


class DocumentContentResponse(BaseModel):
    id: UUID
    title: str
    authors: str | None
    abstract: str | None
    content: str
    status: str
    publication_date: date | None

    model_config = {"from_attributes": True}

    @field_serializer("id")
    def serialize_uuid(self, v: UUID) -> str:
        return str(v)

    @field_serializer("publication_date")
    def serialize_date(self, v: date | None) -> str | None:
        return v.isoformat() if v else None
